src/odemis/dataio/tiff.py

```diff
@@ -1954,5 +1716,443 @@ def read_thumbnail(filename):
+            + shape of original DataArrayShadow. The metadata is the metadata of the first
+            DataArrayShadow of the list
+        """
+        imset = numpy.empty(data_array_shadow.shape, data_array_shadow.dtype)
+        for tiff_info_item in tiff_info:
+            image = AcquisitionDataTIFF._readImage(tiff_info_item['handle'], tiff_info_item['dir_index'])
+            imset[tiff_info_item['hdim_index']] = image
+
+        return model.DataArray(imset, metadata=data_array_shadow.metadata)
+
+    def getData(self, n):
+        """
+        Fetches the whole data (at full resolution) of image at index n.
+        n (0<=int): index of the image
+        return DataArray: the data, with its metadata (ie, identical to .content[n] but
+            with the actual data)
+        """
+        tiff_info = self.content[n].tiff_info
+        # if tiff_info is a list, it means that self.content[n]
+        # is a DataArrayShadow with multiple images
+        if type(tiff_info) is list:
+            return AcquisitionDataTIFF._readAndMergeImages(self.content[n], tiff_info)
+        else:
+            image = AcquisitionDataTIFF._readImage(tiff_info['handle'], tiff_info['dir_index'])
+            return model.DataArray(image, metadata=self.content[n].metadata)
+
+    def getSubData(self, n, z, rect):
+        """
+        Fetches a part of the data, for a given zoom. If the (complete) data has more
+        than two dimensions, all the extra dimensions (ie, non-spatial) are always fully
+        returned for the given part.
+        n (int): index of the image
+        z (0 <= int) : zoom level. The data returned will be with MD_PIXEL_SIZE * 2^z.
+            So 0 means to use the highest zoom, with the original pixel size. 1 will
+            return data half the width and heigh (The maximum possible value depends
+            on the data).
+        rect (4 ints): left, top, right, bottom coordinates (in px, at zoom=0) of the
+            area of interest.
+        return (tuple of tuple of DataArray): all the tiles in X&Y dimension, so that
+            the area of interest is fully covered (so the area can be larger than requested).
+            The first dimension is X, and second is Y. For example, if returning 3x7 tiles,
+            the most bottom-right tile will be accessed as ret[2][6]. For each
+            DataArray.metadata, MD_POS and MD_PIXEL_SIZE are updated appropriately
+            (if MD_POS is not present, (0,0) is used as default for the entire image, and if
+            MD_PIXEL_SIZE is not present, it will not be updated).
+        raise ValueError: if the area or z is out of range, or if the raw data is not pyramidal.
+        """
+        if not (0 <= n < len(self.content)):
+            raise ValueError("Invalid N value %d" % (n,))
+
+        # get information about how to retrieve the actual pixels from the TIFF file
+        tiff_info = self.content[n].tiff_info
+        tiff_file = tiff_info['handle']
```

Comment:
How hard would it be to handle tiff_info when it's a list? At least, add a big "TODO" explaining what would need to be done to handle the extra dimensions.